modules/extensions-spring-support/src/test/java/org/mule/module/extension/internal/metadata/OperationMetadataTestCase.java

```diff
@@ -258,4 +265,36 @@ public void dynamicOutputWithoutKeyParam() throws Exception
 
         assertThat(metadataDescriptor.getParametersMetadata(), empty());
     }
+
+    @Test
+    public void dynamicOutputAndContentWithCache() throws Exception
+    {
+        processorId = new ProcessorId(CONTENT_AND_OUTPUT_CACHE_RESOLVER, FIRST_PROCESSOR_INDEX);
+        final OperationMetadataDescriptor metadataDescriptor = getOperationDynamicMetadata(nullMetadataKey);
+
+        assertThat(metadataDescriptor.getContentMetadata().get().getType(),
+                   is(equalTo(metadataDescriptor.getOutputMetadata().getPayloadMetadata().getType())));
+
+    }
+
+    @Test
+    public void multipleCaches() throws Exception
+    {
+        // using the base config
+        processorId = new ProcessorId(CONTENT_AND_OUTPUT_CACHE_RESOLVER, FIRST_PROCESSOR_INDEX);
+        getOperationDynamicMetadata(nullMetadataKey);
+        processorId = new ProcessorId(OUTPUT_METADATA_WITHOUT_KEY_PARAM, FIRST_PROCESSOR_INDEX);
+        getOperationDynamicMetadata(nullMetadataKey);
+
+        // using the alternative-config
+        processorId = new ProcessorId(CONTENT_AND_OUTPUT_CACHE_RESOLVER_WITH_SPECIFIC_CONFIG, FIRST_PROCESSOR_INDEX);
+        getOperationDynamicMetadata(nullMetadataKey);
+
+        MuleMetadataManager metadataManager = (MuleMetadataManager) muleContext.getRegistry().lookupObject(MetadataManager.class);
+        Map<String, ? extends MetadataCache> caches = metadataManager.getCaches();
+
+        assertThat(caches.keySet(), hasSize(2));
+        assertThat(caches.keySet(), hasItems("config", "alternative-config"));
+    }
```

Comment:
do we have a test to verify that the correct keys are being stored in each cache? this will be even more important if/when we add a persistent implementation.